### PR TITLE
Scaleway provider

### PR DESCRIPTION
**Pull request by rguichard**

### CODE REVIEW

File: modules/scaleway/nginx-ingress.tf [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+      default_network_policy = true
+      ingress_cidr           = "0.0.0.0/0"
+      chart_version          = "2.15.0"
+      version                = "0.35.0"
+      allowed_cidrs          = ["0.0.0.0/0"]
+    },
+    var.nginx_ingress
+  )
+
+  values_nginx_ingress_l4 = <<VALUES
```
Comment (ArchiFleKs) [inline modules/scaleway/nginx-ingress.tf:20]: add service annotation as per: https://github.com/kubernetes/ingress-nginx/blob/master/deploy/static/provider/scw/deploy.yaml#L277

File: .github/workflows/terraform-scaleway.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    - name: Terraform Format
+      run: terraform fmt -check
+      working-directory: modules/scaleway
+
+    - name: Terraform Validate
+      run: terraform validate
+      working-directory: modules/scaleway
+      env:
+        SCW_DEFAULT_ZONE: fr-par-1
```
Comment (ArchiFleKs) [inline .github/workflows/terraform-scaleway.yaml:33]: Are all these variables needed to make validate work locally ?

File: modules/scaleway/examples/terraform/main.tf [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+  nginx_ingress = {
+    enabled = true
+  }
+
+  istio_operator = {
+    enabled = true
+  }
+
+  cluster_autoscaler = {
```
Comment (ArchiFleKs) [inline modules/scaleway/examples/terraform/main.tf:12]: remove cluster autoscaler variable as it is not supported

File: prometheus-operator.tf [hunk 4]
```diff
@@ -6,10 +6,9 @@ locals {
       namespace              = "monitoring"
       chart                  = "prometheus-operator"
       repository             = "https://prometheus-community.github.io/helm-charts"
-      kiam_allowed_regexp    = "^$"
       enabled                = false
       chart_version          = "v9.3.1"
-      allowed_cidr           = "0.0.0.0/0"
+      allowed_cidr           = ["0.0.0.0/0"]
```
Comment (ArchiFleKs) [inline prometheus-operator.tf:11]: allowed_cidrs

File: prometheus-operator.tf [hunk 5]
```diff
@@ -206,7 +201,7 @@ resource "kubernetes_network_policy" "prometheus_operator_allow_control_plane" {
       }
 
       dynamic "from" {
-        for_each = local.prometheus_operator["allowed_cidrs"]
+        for_each = local.prometheus_operator["allowed_cidr"]
```
Comment (ArchiFleKs) [inline prometheus-operator.tf:204]: allowed_cidrs

File: modules/scaleway/versions.tf [hunk 6]
```diff
@@ -0,0 +1,13 @@
+terraform {
+  required_version = ">= 0.13"
+  required_providers {
+    scaleway = {
+      source = "scaleway/scaleway"
+      version = ">= 1.17.0"
+    }
+    kubectl = {
```
Comment (ArchiFleKs) [inline modules/scaleway/versions.tf:8]: remove kubectl provider

File: modules/scaleway/versions.tf [hunk 7]
```diff
@@ -0,0 +1,13 @@
+terraform {
+  required_version = ">= 0.13"
+  required_providers {
+    scaleway = {
+      source = "scaleway/scaleway"
+      version = ">= 1.17.0"
```
Comment (ArchiFleKs) [inline modules/scaleway/versions.tf:6]: version = "~> 1.0"

File: modules/scaleway/variables.tf [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+  default     = {}
+}
+
+variable "istio_operator" {
+  description = "Customize istio operator deployment, see `istio_operator.tf` for supported values"
+  type        = any
+  default     = {}
+}
+
+variable "calico" {
```
Comment (ArchiFleKs) [inline modules/scaleway/variables.tf:108]: to remove

**Review by ArchiFleKs — CHANGES_REQUESTED** [review]

**Comment by ArchiFleKs** [comment]
Rebase and then:

* `cluster-autoscaler.tf` : should be removed
* `kong.tf`: should be linked with `../../kong.tf`
* `locals.tf` should be linked with `../../locals.tf`
* `metrics-server.tf` should be linked with `../../metrics-server.tf`
* `priority-class.tf` -> should be linked with `../../priority-class.tf`
* `promtheus-operator.tf` should be renamed and linked to `../../kube-prometheus-stack.tf`

File: modules/scaleway/nginx-ingress.tf [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+VALUES
+
+}
+
+resource "kubernetes_namespace" "nginx_ingress" {
+  count = local.nginx_ingress["enabled"] ? 1 : 0
+
+  metadata {
+    labels = {
+      name = local.nginx_ingress["namespace"]
```
Comment (ArchiFleKs) [inline modules/scaleway/nginx-ingress.tf:49]: "${local.labels_prefix}/component" = "ingress"

**Review by ArchiFleKs — CHANGES_REQUESTED** [review]

File: modules/scaleway/versions.tf (hunk 7 above)
Comment (rguichard) [reply modules/scaleway/versions.tf:6]: there is some breaking changes starting from `1.11`.

Comment (ArchiFleKs) [reply modules/scaleway/versions.tf:6]: Then they should have make it 2.0 :p

Comment (rguichard) [reply modules/scaleway/versions.tf:6]: Not my provider :P

File: modules/scaleway/external-dns.tf [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+    ingress {
+      ports {
+        port     = "http"
+        protocol = "TCP"
+      }
+
+      from {
+        namespace_selector {
+          match_labels = {
+            name = kubernetes_namespace.kube-prometheus-stack.*.metadata.0.name[count.index]
```
Comment (ArchiFleKs) [inline modules/scaleway/external-dns.tf:148]: "${local.labels_prefix}/component" = "monitoring" instead

File: modules/scaleway/flux.tf [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+    ingress {
+      ports {
+        port     = "3030"
+        protocol = "TCP"
+      }
+
+      from {
+        namespace_selector {
+          match_labels = {
+            name = kubernetes_namespace.kube-prometheus-stack.*.metadata.0.name[count.index]
```
Comment (ArchiFleKs) [inline modules/scaleway/flux.tf:177]: "${local.labels_prefix}/component" = "monitoring"

File: modules/scaleway/ingress-nginx.tf [hunk 12]
```diff
@@ -0,0 +1,236 @@
...
+    ingress {
+      ports {
+        port     = "metrics"
+        protocol = "TCP"
+      }
+
+      from {
+        namespace_selector {
+          match_labels = {
+            name = kubernetes_namespace.kube-prometheus-stack.*.metadata.0.name[count.index]
```
Comment (ArchiFleKs) [inline modules/scaleway/ingress-nginx.tf:191]: "${local.labels_prefix}/component" = "monitoring"

**Review by ArchiFleKs — CHANGES_REQUESTED** [review]

File: modules/scaleway/README.md [hunk 13]
```diff
@@ -0,0 +1,72 @@
+# terraform-kubernetes-addons
+
+[![Build Status](https://github.com/particuleio/terraform-kubernetes-addons/workflows/Terraform/badge.svg)](https://github.com/particuleio/terraform-kubernetes-addons/actions?query=workflow%3ATerraform)
```
Comment (ArchiFleKs) [inline modules/scaleway/README.md:3]: s/Terraform/terraform:scaleway

**Review by ArchiFleKs — CHANGES_REQUESTED** [review]

**Pull request merged**